std/csv.d

```diff
@@ -1750,3 +1769,19 @@ if (isSomeChar!Separator && isInputRange!Range
         ++i;
     }
 }
+
+// https://issues.dlang.org/show_bug.cgi?id=21629
+@safe pure unittest
+{
+    struct Reccord
+    {
+        string a;
+        string b;
+    }
+
+    string input = "";
+    foreach (it; csvReader!Reccord(input))
+    {
+        assert(false, "We shouldn't make it here");
```

Comment:
Maybe you could count loop iterration and assert the count is zero so that the linter can be happy.

Reply:
done